scalar.go

```diff
@@ -31,21 +32,48 @@ type Identifier struct {
 }
 
 type IdentifierInput struct {
```

Comment:
The identifier input's `id` and `alias` can now only be read through `input.Id()` and `input.Alias()`. As a result, `IdentifierInput` can no longer be mutated. 

This change is beneficial because:

1. It forces end users to use the `NewIdentifier()` function everywhere rather than instantiating identifiers on their own.
2. It prevents users from making malformed `IdentifierInputs` (for example one where there is no `id` and no `alias`, or the `id` and `alias` fields are swapped, or both `id` and `alias` are set.)

By locking down the `IdentifierInput` we can also create a special type to be used for `ServiceUpdateInput` to unset optional identifier fields, by editing the `MarshalJSON()` logic.